src/components/nav-icon-item/nav-icon-item.tsx

```diff
@@ -0,0 +1,63 @@
+import { AnchorHTMLAttributes, ButtonHTMLAttributes, FC, MouseEventHandler, ReactNode } from 'react'
+import { ElAnchorNavIconItem, ElButtonNavIconItem } from './styles'
+
+interface CommonNavIconItemProps {
+  /**
+   * The JSX element to render as the icon
+   */
+  icon: ReactNode
+
+  /**
+   * Whether the nav item is active
+   * @default false
+   **/
+  isActive?: boolean
+}
+
+interface NavIconItemAsButtonElementProps extends CommonNavIconItemProps, ButtonHTMLAttributes<HTMLButtonElement> {
+  href?: never
+  target?: never
+  ref?: never
+  onClick: MouseEventHandler<HTMLButtonElement>
+}
+
+interface NavIconItemAsAnchorElementProps extends CommonNavIconItemProps, AnchorHTMLAttributes<HTMLAnchorElement> {
+  href?: string
+  target?: string
+  rel?: string
+  onClick?: MouseEventHandler<HTMLAnchorElement>
+}
+
+export type NavIconItemProps = NavIconItemAsButtonElementProps | NavIconItemAsAnchorElementProps
+
+const isButtonElement = (props: NavIconItemProps): props is NavIconItemAsButtonElementProps => {
+  return !props.href
+}
+
+export const NavIconItem: FC<NavIconItemProps> = (props) => {
+  if (!isButtonElement(props)) {
```

Comment:
**quibble:** for readability, I personally find it best to keep simple if-else blocks like this one written so that the if-condition is _not_ negated. That is, I find

```javascript
if (isButtonElement(props)) {
  // a button element
} else {
  // not a button element
}
```

Alternatively, changing `isButtonElement` to `isAnchorElement` would also make it more readable, because our if-condition does not involve a negation.

Reply:
> for readability, I personally find it best to keep simple if-else blocks like this one written so that the if-condition is not negated

placing the truthy condition based on the function name makes perfect sense